src/consensus/aft/raft.h

```diff
@@ -3333,18 +3425,26 @@ namespace aft
           }
           else
           {
+            LOG_TRACE_FMT(
+              "Configurations: not enough trusted or retired nodes for "
+              "configuration #{} ({}/{} trusted, {}/{} retired)",
+              next->rid,
+              num_trusted_nodes,
+              next->nodes.size(),
+              num_retired_nodes,
+              num_required_retired_nodes);
             if (
-              use_two_tx_reconfig && !is_learner() && !is_retired() &&
-              node_client &&
-              next->nodes.find(state->my_node_id) != next->nodes.end())
+              node_client && !is_learner() && !is_retired() &&
+              ((next->nodes.find(state->my_node_id) != next->nodes.end()) ||
+               (is_retiring() &&
+                next->nodes.find(state->my_node_id) == next->nodes.end())))
             {
-              LOG_TRACE_FMT(
-                "Configurations: not enough trusted nodes for configuration "
-                "#{} ({} out of {}); submitting ORC",
+              schedule_submit_orc(
```

Comment:
You mentioned in our discussion that we stop scheduling the ORC when we see our `RETIRED` state as committed. It seems that the only condition to stop scheduling the ORC is when an ORC has been forwarded to a node (`submit_orc()` returns true in `orc_requests.h`), which isn't enough to guarantee that the ORC has successfully been applied (the request may have been lost, or the transaction rolled back). What am I missing?

Reply:
We do need to keep retrying until we've observed the ORC committed or our transition complete (retired or trusted), a node having accepted (to forward) the RPC isn't enough.

Reply:
It will keep scheduling new ORCs on every `commit()` _until_ it becomes `RETIRED` (which also means that the entire rest of the new configuration is ready to be applied, so no more ORCs needed). But, we don't need to know whether the ORC RPC call itself succeeded - we can't do anything to correct the error anyways, all we can do is retry. Am I missing something?

Reply:
https://github.com/microsoft/CCF/pull/3097#discussion_r741986320